### PR TITLE
basic SSR support added

### PR DESCRIPTION
**Pull request by szagi3891**

### CODE REVIEW

File: crates/vertigo-cli/src/serve/mount_path.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+    None
+}
+
+pub fn find_wasm(nodes: &Arc<Vec<Node>>) -> Option<String> {
+    find_wasm_nodes(nodes.as_slice())
+}
+
+fn replace_prefix(public_path: &str, dest_dir: &str, path: &str) -> Result<String, i32> {
+    let Some(rest) = path.strip_prefix(public_path) else {
```
Comment (sfisol) [inline crates/vertigo-cli/src/serve/mount_path.rs:131]: Operating on `Path` object will be safer here

File: crates/vertigo-cli/src/serve/mount_path.rs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        log::error!("Incorrect path http: path={path} (public_path={public_path}, dest_dir={dest_dir})");
+        return Err(-1);
+    };
+
+    Ok(format!("{dest_dir}{rest}"))
```
Comment (sfisol) [inline crates/vertigo-cli/src/serve/mount_path.rs:136]: Then here it can be `dest_dir.join(rest)`

**Review by sfisol — CHANGES_REQUESTED** [review]

File: crates/vertigo-cli/src/serve/html/dom_command.rs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    "#;
+
+    let commands = serde_json::from_str::<Vec<DomCommand>>(data).unwrap();
+
+    assert_eq!(commands.len(), 2);
+    assert_eq!(commands[0], DomCommand::CreateNode { id: 2, name: "div".into() });
+    assert_eq!(commands[1], DomCommand::CallbackAdd { id: 2, event_name: "hook_keydown".into(), callback_id: 2 } );
+}
+
+#[test]
```
[the remark this answers is not in the document]
Comment (szagi3891) [reply crates/vertigo-cli/src/serve/html/dom_command.rs]: done

File: crates/vertigo-cli/src/serve/html/ordered_map.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        for key in self.order.iter() {
+            let value = self.data.get(key).unwrap();
+            result.push((key.clone(), value.clone()));
+        }
+
+        result
+    }
+}
+
+#[test]
```
[the remark this answers is not in the document]
Comment (szagi3891) [reply crates/vertigo-cli/src/serve/html/ordered_map.rs:49]: done

**Review by sfisol — APPROVED** [review]

**Pull request merged**